experimental/zig/patches/nostrip.patch

```diff
@@ -0,0 +1,12 @@
+we do this ourselves later
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -835,7 +835,7 @@
+ elseif("${CMAKE_BUILD_TYPE}" STREQUAL "RelWithDebInfo")
+   set(ZIG_RELEASE_ARG -Doptimize=ReleaseFast)
+ else()
+-  set(ZIG_RELEASE_ARG -Doptimize=ReleaseFast -Dstrip)
```

Comment:
This also affects conditional compilation & code generation - stack trace printing code will be omitted when strip is true. If you strip later, this code will remain, bloating the binary, but unable to be useful.

Reply:
ah, good to know. i assume the fact it's not actually strip(1) but rather objcopy'd with debuglink (-dbg subpackaged) still makes it useless? (i.e. zig itself won't be able to load it, so may as well just strip it fully)

Reply:
Hmm well it may be the case that zig could be enhanced to find -dbg files being separately packaged. But let me switch topics for a moment. You shouldn't need that patch because if you want debug info you should use `-DCMAKE_BUILD_TYPE=RelWithDebInfo` as you can see in the if statement above. I would expect this to already be the default in chimera's package manager. Not sure why you would need a patch here.

Reply:
the default is [=None](https://github.com/chimera-linux/cports/blob/46fe927aff7a3e98a52f8f2b83f4dbf1a9f36a65/src/cbuild/util/cmake.py#L92), aside from this conditional the only difference for cmake is `RelWithDebInfo` appends `-O2 -g` to c*flags (None does nothing), and in this case everything is identical regardless of build type up until this conditional

Reply:
completely offtopic, the general philosophical reasoning distros use =None instead of other types is because aside from setting some default flags appended to env flags, they don't actually do anything by themselves. i.e.:

```
None:
Release: -O3 -DNDEBUG
RelWithDebInfo: -O2 -g
MinSizeRel: -Os -DNDEBUG
```

and they are appended to the env (`CFLAGS="stuff" cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo` gives `CFLAGS="stuff -O2 -g"`), which is sometimes unwanted. this basic difference aside, it's project-dependent what anyone does (sometimes a non-release type adds a bunch of stuff one doesn't want, sometimes the *release* type adds a bunch of stuff one doesn't want (in contrast to None), ...)

it's really a case by case basis so i read through every cmake project because people end up doing anything imaginable with conditionals :D

Reply:
Thanks for the explanation! That is a useful perspective for me to be aware of.